test/integration/client-side-encryption/driver.test.ts

```diff
@@ -0,0 +1,297 @@
+          f: new BSON.BSONRegExp('[A-Za-z0-9]*'),
+          g: undefined
+        };
+
+        const expected = BSON.deserialize(BSON.serialize(data, bsonOptions), bsonOptions);
+
+        const coll = encryptedClient.db(dataDbName).collection(dataCollName);
+        const result = await coll.insertOne(data, bsonOptions);
+        const actual = await coll.findOne({ _id: result.insertedId }, bsonOptions);
+        const gValue = actual?.g;
+        delete actual?.g;
+
+        expect(actual).to.deep.equal(expected);
+        expect(gValue).to.equal(bsonOptions.ignoreUndefined ? data.g : null);
+      });
+    }
+  });
+
+  describe('key order aware command properties', () => {
+    let client: MongoClient;
+    let collection: Collection;
+
+    beforeEach(async function () {
+      if (!this.configuration.clientSideEncryption.enabled) {
+        return;
+      }
+
+      const encryptionOptions = {
+        monitorCommands: true,
+        autoEncryption: {
+          keyVaultNamespace,
+          kmsProviders: { local: { key: 'A'.repeat(128) } }
+        }
+      };
+      client = this.configuration.newClient({}, encryptionOptions);
+      collection = client.db(dataDbName).collection('keyOrder');
+    });
+
+    afterEach(async () => {
+      if (client) await client.close();
+    });
+
+    describe('find', () => {
+      it('should maintain ordered sort', metadata, async function () {
+        const events: CommandStartedEvent[] = [];
+        client.on('commandStarted', ev => events.push(ev));
+        const sort: ReadonlyArray<[string, SortDirection]> = Object.freeze([
+          ['1', 1],
+          ['0', 1]
+        ]);
+        await collection.findOne({}, { sort });
+        const findEvent = events.find(event => !!event.command.find);
+        expect(findEvent).to.have.property('commandName', 'find');
+        expect(findEvent).to.have.nested.property('command.sort').deep.equal(new Map(sort));
+      });
+    });
+
+    describe('findAndModify', () => {
+      it('should maintain ordered sort', metadata, async function () {
+        const events: CommandStartedEvent[] = [];
+        client.on('commandStarted', ev => events.push(ev));
+        const sort: [string, SortDirection][] = [
+          ['1', 1],
+          ['0', 1]
+        ];
+        await collection.findOneAndUpdate({}, { $setOnInsert: { a: 1 } }, { sort });
+        const findAndModifyEvent = events.find(event => !!event.command.findAndModify);
+        expect(findAndModifyEvent).to.have.property('commandName', 'findAndModify');
+        expect(findAndModifyEvent)
+          .to.have.nested.property('command.sort')
+          .deep.equal(new Map(sort));
+      });
+    });
+
+    describe('createIndexes', () => {
+      it('should maintain ordered index keys', metadata, async function () {
+        const events: CommandStartedEvent[] = [];
+        client.on('commandStarted', ev => events.push(ev));
+        const indexDescription: [string, IndexDirection][] = [
```

Comment:
pointing out this case since you only addressed the freeze in the `find` function test case

Reply:
fixed